src/main/java/org/junitpioneer/jupiter/resource/ResourceExtension.java

```diff
@@ -323,7 +332,37 @@ public <T> T interceptTestClassConstructor(Invocation<T> invocation,
 	@Override
 	public void interceptBeforeAllMethod(Invocation<Void> invocation,
 			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
-		runSequentially(invocation, invocationContext.getExecutable(), extensionContext);
+		Class<?> declaringClass = invocationContext.getExecutable().getDeclaringClass();
+		List<Class<?>> nestedClasses = findNestedClassesRecursively(declaringClass);
+		Set<Shared> sharedAnnotations = Stream
+				.concat(Stream.of(declaringClass), nestedClasses.stream())
+				.flatMap(clazz -> ReflectionSupport
+						.findMethods(clazz, unused -> true, HierarchyTraversalMode.TOP_DOWN)
```

Comment:
Should be using [AnnotationSupport::findAnnotatedMethods](https://junit.org/junit5/docs/5.8.1/api/org.junit.platform.commons/org/junit/platform/commons/support/AnnotationSupport.html#findAnnotatedMethods(java.lang.Class,java.lang.Class,org.junit.platform.commons.support.HierarchyTraversalMode))

Reply:
I can see why you think that! But sadly it doesn't work in this case.

It can find methods like this:

```java
@Shared
@Test
public void someTest() { ... }
```

but not like this:

```java
@Test
public void someTest(@Shared parameter) { ... }
```

Hence the need to grab all methods, then filter over the ones with `@Shared`-annotated parameters.

Reply:
@Michael1993 I've found a way to eliminate the "heisenbug" in this area of the code, so this code no longer exists. Thank you very much for your thoughts anyway. :)